app/navigation/CreateNavigator.scala

```diff
@@ -0,0 +1,235 @@
+    P(IndirectRepresentativePage, controllers.routes.IndirectRepresentativeController.onPageLoad, showIndirectRepresentative, indirectRepresentativeAnswered),
+    P(ProofOfAuthorityPage, controllers.routes.ProofOfAuthorityController.showFileUpload, showProofOfAuthority, proofOfAuthorityAnswered),
+
+    P(BankDetailsPage, controllers.routes.BankDetailsController.onPageLoad, showBankDetails, bankDetailsAnswered),
+
+    P(CheckYourAnswersPage, controllers.routes.CheckYourAnswersController.onPageLoad, always, never),
+    P(ConfirmationPage, controllers.routes.ConfirmationController.onPageLoad, always, never)
+  )
+  // @formatter:off
+}
+
+protected trait CreateAnswerConditions {
+
+  protected val always: UserAnswers => Boolean = (_: UserAnswers) => true
+
+  protected val isImporter: UserAnswers => Boolean = _.isImporterJourney
+  protected val isAgent: UserAnswers => Boolean    = _.isAgentJourney
+
+  protected val showUkRegulationType: UserAnswers => Boolean = (answers: UserAnswers) =>
+    answers.get(CustomsRegulationTypePage).contains(CustomsRegulationType.UKCustomsCodeRegulation)
+
+  protected val showArticleType: UserAnswers => Boolean = (answers: UserAnswers) =>
+    answers.get(CustomsRegulationTypePage).contains(CustomsRegulationType.UnionsCustomsCodeRegulation)
+
+  protected val showBulkUpload: UserAnswers => Boolean = (answers: UserAnswers) =>
+    answers.isMultipleEntry
+
+  protected val showCustomsDutyPaid: UserAnswers => Boolean = (answers: UserAnswers) =>
+    answers.get(ClaimRepaymentTypePage).exists(_.contains(ClaimRepaymentType.Customs))
+
+  protected val showVatPaid: UserAnswers => Boolean = (answers: UserAnswers) =>
+    answers.get(ClaimRepaymentTypePage).exists(_.contains(ClaimRepaymentType.Vat))
+
+  protected val showOtherDutyPaid: UserAnswers => Boolean = (answers: UserAnswers) =>
+    answers.get(ClaimRepaymentTypePage).exists(_.contains(ClaimRepaymentType.Other))
+
+  protected val showFileUpload: UserAnswers => Boolean = (answers: UserAnswers) =>
+    answers.fileUploadState.isEmpty || answers.fileUploadState.exists(state => state.fileUploads.isEmpty)
+
+  protected val showImporterEori: UserAnswers => Boolean = (answers: UserAnswers) =>
+    answers.isImporterJourney && answers.get(ImporterHasEoriPage).contains(true)
+
+  protected val showImporterAgentEori: UserAnswers => Boolean = (answers: UserAnswers) =>
+    answers.isAgentJourney && answers.get(ImporterHasEoriPage).contains(true)
+
+  protected val showEnterAgentEori: UserAnswers => Boolean = (answers: UserAnswers) =>
+    answers.isAgentJourney && answers.get(AgentImporterHasEORIPage).contains(AgentImporterHasEORI.Yes)
+
+  protected val showImporterName: UserAnswers => Boolean = (answers: UserAnswers) =>
+    answers.isImporterJourney && answers.get(DoYouOwnTheGoodsPage).contains(DoYouOwnTheGoods.No)
+
+  protected val showRepaymentType: UserAnswers => Boolean = (answers: UserAnswers) =>
+    answers.isSingleEntry
```

Comment:
I am sure this is right but I am curious.
why does isSingleEntry decide if we showRepaymentType

Reply:
Logic in this navigator is pretty much determined by analysing logic in original navigator.   Old navigator works by deciding what page to go to next (rather than what condition(s) make this page required).  So I generally looked for places where the old navigator redirected to this page.   In this case, there was only one place.